instrumentation/kafka-streams/src/test/java/brave/kafka/streams/KafkaStreamsTracingTest.java

```diff
@@ -1,28 +1,176 @@
+    kafkaStreamsTracing.nextSpan(fakeProcessorContext).start().finish();
+
+    assertThat(spans)
+            .flatExtracting(s -> s.tags().entrySet())
+            .containsOnly(
+                    entry("kafka.streams.application.id", TEST_APPLICATION_ID),
+                    entry("kafka.streams.task.id", TEST_TASK_ID));
+  }
+
+  @Test
+  public void processorSupplier_should_tag_key_if_string() {
+    Processor<String, String> processor = fakeProcessorSupplier.get();
+    processor.init(processorContextSupplier.apply(new RecordHeaders()));
+    processor.process(TEST_KEY, TEST_VALUE);
+
+    assertThat(spans)
+            .flatExtracting(s -> s.tags().entrySet())
+            .containsOnly(
+                    entry("kafka.streams.application.id", TEST_APPLICATION_ID),
+                    entry("kafka.streams.task.id", TEST_TASK_ID),
+                    entry("kafka.streams.key", TEST_KEY));
+  }
+
+  @Test
+  public void processorSupplier_should_tag_key_if_null() {
+    Processor<String, String> processor = fakeProcessorSupplier.get();
+    processor.init(processorContextSupplier.apply(new RecordHeaders()));
+    processor.process(null, TEST_VALUE);
+
+    assertThat(spans)
+            .flatExtracting(s -> s.tags().entrySet())
+            .containsOnly(
+                    entry("kafka.streams.application.id", TEST_APPLICATION_ID),
+                    entry("kafka.streams.task.id", TEST_TASK_ID));
+  }
+
+  @Test
+  public void processorSupplier_should_tag_key_if_binary() {
+    ProcessorSupplier<byte[], String> fakeProcessorSupplier =
+            kafkaStreamsTracing.processorSupplier(
+                    "processor-1",
+                    new AbstractProcessor<byte[], String>() {
+                      @Override
+                      public void process(byte[] key, String value) {
+                        context().forward(key, value);
+                      }
+                    });
+
+    Processor<byte[], String> processor = fakeProcessorSupplier.get();
+    processor.init(processorContextSupplier.apply(new RecordHeaders()));
+    processor.process(new byte[1], TEST_VALUE);
+
+    assertThat(spans)
+            .flatExtracting(s -> s.tags().entrySet())
+            .containsOnly(
+                    entry("kafka.streams.application.id", TEST_APPLICATION_ID),
+                    entry("kafka.streams.task.id", TEST_TASK_ID));
+  }
+
+  @Test
+  public void transformSupplier_should_tag_key_if_string() {
+    Transformer processor = fakeTransformerSupplier.get();
+    processor.init(processorContextSupplier.apply(new RecordHeaders()));
+    processor.transform(TEST_KEY, TEST_VALUE);
+
+    assertThat(spans)
+            .flatExtracting(s -> s.tags().entrySet())
+            .containsOnly(
+                    entry("kafka.streams.application.id", TEST_APPLICATION_ID),
+                    entry("kafka.streams.task.id", TEST_TASK_ID),
+                    entry("kafka.streams.key", TEST_KEY));
```

Comment:
is key usually a good thing to know? not sure if it is or not just asking

Reply:
Could be if Processor/Transformer is used in a "non-traced" Topology, where `KafkaClientSupplier` is not used, and it just creates a individual `span` with the `key` available as a tag. I should test this scenario.